src/math/p5.Vector.js

```diff
@@ -118,23 +127,225 @@ function vector(p5, fn) {
     // This is how it comes in with createVector()
     // This check if the first argument is a function
     constructor(...args) {
-      let x, y, z;
+      let dimensions = args.length; // TODO: make default 3 if no arguments
+      let values = args.map(arg => arg || 0);
       if (typeof args[0] === 'function') {
         this.isPInst = true;
         this._fromRadians = args[0];
         this._toRadians = args[1];
-        x = args[2] || 0;
-        y = args[3] || 0;
-        z = args[4] || 0;
-        // This is what we'll get with new p5.Vector()
+        values = args.slice(2).map(arg => arg || 0);
+      }
+      if (dimensions === 0) {
+        this.dimensions = 2;
+        this._values = [0, 0, 0];
+
+      } else {
+        this.dimensions = dimensions;
+        this._values = values;
+      }
+    }
+
+    /**
+     * Gets the values of the vector.
+     *
+     * This method returns an array of numbers that represent the vector. 
+     * Each number in the array corresponds to a different component of the vector, 
+     * like its position in different directions (e.g., x, y, z).
+     *
+     * @returns {Array<number>} The array of values representing the vector.
+     */
+    get values() {
+      return this._values
+    }
+
+    /**
+     * Sets the values of the vector.
+     *
+     * This method allows you to update the entire vector with a new set of values.
+     * You need to provide an array of numbers, where each number represents a component
+     * of the vector (e.g., x, y, z). The length of the array should match the number of
+     * dimensions of the vector. If the array is shorter, the missing components will be
+     * set to 0. If the array is longer, the extra values will be ignored.
+     *
+     * @param {Array<number>} newValues - An array of numbers representing the new values for the vector.
+     *
+     */
+    set values(newValues) {
+      let dimensions = newValues.length; 
+      if (dimensions === 0) {
+        this.dimensions = dimensions;
```

Comment:
I notice this is a little different than in the constructor, where we set `dimensions` to 2 in this case. Do we want this method to work the same way?

Reply:
yeah it was a mis, will fix, thanks